### ISSUE TITLE
HACS not updating past 1.19.3

### ISSUE THREAD

**Issue by cremor** [hacs/integration#2556]
## System Health

version | core-2022.3.4
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.9.9
os_name | Linux
os_version | 5.10.92-v8
arch | aarch64
timezone | Europe/Vienna

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
Github API Calls Remaining | 4989
Installed Version | 1.19.3
Stage | running
Available Repositories | 1004
Downloaded Repositories | 1

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 7.4
-- | --
update_channel | stable
supervisor_version | supervisor-2022.03.3
docker_version | 20.10.9
disk_total | 118.3 GB
disk_used | 5.2 GB
healthy | true
supported | true
board | rpi4-64
supervisor_api | ok
version_api | ok
installed_addons | File editor (5.3.3), SSH & Web Terminal (10.0.2), Home Assistant Google Drive Backup (0.105.2)

</details>

<details><summary>Lovelace</summary>

dashboards | 1
-- | --
resources | 0
views | 2
mode | storage

</details>

### Checklist

- [X] I'm running the newest version of HACS <https://github.com/hacs/integration/releases/latest>
- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] I have read <https://hacs.xyz/docs/issues>
- [X] This issue is related to the backend (integration part) of HACS.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of currently [open](https://github.com/hacs/integration/issues) or issues [pending release](https://github.com/hacs/integration/issues?q=is%3Aissue+is%3Aclosed+sort%3Aupdated-desc+milestone%3Anext).

### Describe the issue

The first version of HACS I ever installed was 1.21.0. After I noticed that I didn't get any update notifications and also couldn't manually trigger an update, I searched for existing issues and found #2553.
As suggested in this issue, I "reinstalled" HACS by executing the download command again. But this actually **downgraded** my HACS to 1.19.3. And I'm still not getting any updates.

I found a few things that might be related:
- The script from https://get.hacs.xyz downloads https://github.com/hacs/integration/releases/latest/download/hacs.zip which seems to only contain version 1.19.3 (I have no idea how I initially got 1.21.0).
- https://github.com/hacs/integration/releases also lists 1.19.3 as the latest version.
- Only on https://github.com/hacs/integration/tags you can see that there are tags for newer versions.

### Reproduction steps

1. Install HACS via the documented/recommended steps.
2. Notice that version 1.19.3 is installed, although there is already a tag for version 1.23.0
3. Wait for a few hours to maybe get an update notification.
4. Notice that no update notification is shown.
5. Use the "update information" (wording might be different, I had to translate) button in the HACS integration card.
6. Notice that nothing happens.

### Debug logs

```text
https://pastebin.com/rNUVK6aX
```

### Diagnostics dump

Although I use the latest version of HA and did not configure HACS with YAML, I do not have a "Download diagnostics" button on the HACS integration.

**Comment by ludeeus** [comment]
So what are you trying to upgrade?
Your logs does not show you upgrading anything.

**Comment by cremor** [comment]
I'd like to update HACS itself.

**Comment by paolo-ermit** [comment]
Yes, I've the same problem. It seems that HACS can't find any update newer than 1.9.3 (that is, BTW, the same last release reported by github)

**Comment by Nebur692** [comment]
My log:

```
Este error se originó a partir de una integración personalizada.

Logger: custom_components.hacs
Source: custom_components/hacs/tasks/base.py:34
Integration: HACS (documentation, issues)
First occurred: 22:49:53 (1 occurrences)
Last logged: 22:49:53

HacsTask<update_critical_repositories> failed: <Integration custom-components/sensor.custom_aftership> Repository has been requested to be removed.
```

**Comment by sau7777** [comment]
![HACS_–_Home_Assistant](https://user-images.githubusercontent.com/8371994/158349853-07552000-db2e-428d-89c7-acc3a3902274.jpg)

**Comment by lmagyar** [comment]
HACS [releases](https://github.com/hacs/integration/releases) slowly disappearing from Github, a few days ago the last was 1.20.0, than it disappeared. The latest [tag](https://github.com/hacs/integration/tags) is 1.23.0, and that was the latest release a few weeks ago, but meantime 1.23, 1.22, 1.21, 1.20 releases disappeared from Github. Strange.

**Comment by ludeeus** [comment]
The releases are there, but they are hidden from the main release page.
I have opened a ticket with GitHub.

**Comment by lmagyar** [comment]
@sau7777 Duplicate of #2495 that is fixed in 1.23.0.

**Comment by cremor** [comment]
> The releases are there, but they are hidden from the main release page.
> I have opened a ticket with GitHub.

Can you still see the releases? If so, have you checked that they are not marked as draft (or however Github calls it)?

**Comment by ACiDGRiM** [comment]
I can only see release 1.19.x in the latest release.
the get script that pulls the hacs.zip from the "latest" release, also reinstalls 1.19.x, not 1.23

**Comment by ludeeus** [comment]
> Can you still see the releases?
Yeah, <https://github.com/hacs/integration/releases/1.23.0>

**Comment by jonathansheehan** [comment]
I see this same issue.  I can see the releases in github but HACS shows up to date.

**Comment by JesperWe** [comment]
FWIW I am seeing (with supervisor-2022.03.4, core-2022.3.5) 

> You are running version 1.16.0, version 1.19.3 is available

I press "Update", then restart HA. But I am still on 1.16.0 and getting the same message.

(Slight variation from the reported issue but most likely related...)

**Comment by ACiDGRiM** [comment]
> > Can you still see the releases?
> 
> Yeah, https://github.com/hacs/integration/releases/1.23.0

https://github.com/hacs/integration/releases/ it doesn't show up here, however

https://github.com/hacs/integration/releases/latest links to 1.19.3

**Comment by lmagyar** [comment]
Until this issue is resolved, if you want to upgrade HACS to the latest 1.23.0 version, run this (I've forked the install script and [hard wired](https://github.com/lmagyar/get/commit/2703cd39b51fbe4b148db17da0efaefc3e65e4a8) the exact 1.23.0 version instead of the 'latest', tested, works, repo will be deleted after this issue is resolved):

```
wget -O - https://raw.githubusercontent.com/lmagyar/get/main/get | bash -
```

Don't forget to restart core after HACS update!

**Comment by yousaf465** [comment]
same issue can't update any HACS components
https://community-assets.home-assistant.io/original/4X/e/f/8/ef8ceb25d57abe95be3593610450366d2f86da8c.png

**Comment by ACiDGRiM** [comment]
I notice that the 1.22 and 1.23 releases do not have a commit signature listed as "verified" could this be related?

**Comment by ludeeus** [comment]
No, they have that as I pushed directly right before releasing and did not use a PR for the last commit before the tag was created.

**Comment by ludeeus** [comment]
GitHub is still investigating what is happening here, there are a small number of other repositories that experience the same behavior.

For now, all missing releases have been restored in the API (and the release page) so this will now work as it should inside HACS.

For authors that might stumble across this issue and are experiencing the same, the first thing you should do is to open a ticket with GitHub support. Once that is done you can go to the tag list in your repo, open the missing release, click the pencil, and then hit the "update" button without doing any changes, which will relink/restore that release.
